src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexIgnoreCaseTests.cs

```diff
@@ -0,0 +1,221 @@
+            {
+                using (_ = new ThreadCultureChange(culture))
+                {
+                    // This test will try to emit code that looks like: textInfo = CultureInfo.GetCultureInfo(CurrentCulture.Name).TextInfo
+                    // so we will validate in this test that we are able to do that for all cultures and that GetCultureInfo returns a valid Culture.
+                    _ = await RegexHelpers.GetRegexAsync(RegexEngine.SourceGenerated, @"(.)\1", RegexOptions.IgnoreCase);
+                }
+            }
+        }
+
+        // This test takes a long time to run since it needs to compute all possible lowercase mappings across
+        // 3 different cultures and then creates Regex matches for all of our engines for each mapping.
+        [OuterLoop]
+        [Theory]
+        [MemberData(nameof(Unicode_IgnoreCase_TestData))]
+        public async Task Unicode_IgnoreCase_Tests(RegexEngine engine, string culture, RegexOptions options)
+        {
+            var testCases = GetPatternAndInputsForCulture(culture);
+
+            foreach ((string pattern, string input) in testCases)
+            {
+
+                if ((options & RegexOptions.CultureInvariant) == 0)
+                {
+                    using var _ = new ThreadCultureChange(culture);
+                    await ValidateMatch(pattern, input);
+                    await ValidateMatch(input, pattern);
+                }
+                else
+                {
+                    await ValidateMatch(pattern, input);
+                    await ValidateMatch(input, pattern);
+                }
+            }
+
+            return;
+
+            async Task ValidateMatch(string pattern, string input)
+            {
+                Regex regex = await RegexHelpers.GetRegexAsync(engine, pattern, options | RegexOptions.IgnoreCase);
+                Assert.True(regex.IsMatch(input));
+            }
+
+            IEnumerable<(string, string)> GetPatternAndInputsForCulture(string culture)
+            {
+                TextInfo textInfo = string.IsNullOrEmpty(culture) ? CultureInfo.InvariantCulture.TextInfo
+                    : CultureInfo.GetCultureInfo(culture).TextInfo;
+
+                for (int i = 0; i < 0x1_0000; i++)
+                {
+                    char c = (char)i;
+                    char lowerC = textInfo.ToLower(c);
+                    if (c != lowerC)
+                    {
+                        yield return ($"{c}", $"{lowerC}");
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> Unicode_IgnoreCase_TestData()
+        {
+            foreach (string culture in new[] { "", "EN-US", "tr-TR", "AZ" })
+            {
+                foreach (RegexEngine engine in RegexHelpers.AvailableEngines)
+                {
+                    // Source generated engine does not support setting different culutres yet.
```

Comment:
```suggestion
                    // Source generated engine does not support setting different cultures yet.
```

That said... I'd have expected this to "just work".  The source generator doesn't end up respecting CultureInfo.CurrentCulture?